src/main/java/com/scottlogic/AuthorPostSorter.java

```diff
@@ -0,0 +1,24 @@
+package com.scottlogic;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class AuthorPostSorter implements PostSorter {
+
+    public List<UserPost> sort(List<UserPost> inputList, SortOrder sortDirection) {
+        if (inputList == null || inputList.size() == 0) {
+            return Arrays.asList();
+        }
+
+        if (sortDirection == SortOrder.ASC) {
```

Comment:
Minor point:
I do believe == works here, but since SortOrder is an enum, it might be clearer or more provide more confidence to use SortOrder.ASC.equals().

For the same reason, with string comparisons on Java, we tend to do "something".equals(someStr) over ==